world/handlers.go

```diff
@@ -199,13 +201,25 @@ func (server *Server) handleChangeRate(conn mnet.Server, reader mpacket.Reader)
 
 	switch mode {
 	case 1:
-		server.info.Rates.Exp = float32(rs)
+		server.Info.Rates.Exp = float32(rs)
 	case 2:
-		server.info.Rates.Drop = float32(rs)
+		server.Info.Rates.Drop = float32(rs)
 	case 3:
-		server.info.Rates.Mesos = float32(rs)
+		server.Info.Rates.Mesos = float32(rs)
 	}
 
+	h := sha512.New()
+	h.Write([]byte(fmt.Sprintf("exp %.2f drop %.2f mesos %.2f", server.Info.Rates.Exp, server.Info.Rates.Drop, server.Info.Rates.Mesos)))
+	hRates := hex.EncodeToString(h.Sum(nil))
+
+	if hRates != server.Info.DefaultRatesChecksum { // Rates event
+		server.Info.Ribbon = 1
```

Comment:
Have you experimented with this byte in the login server? https://github.com/Hucaru/Valhalla/blob/master/login/packets.go#L203

Reply:
Changed it to 1 and nothing happend, will try different values